### PR TITLE
Added config settings to specify locations of individual log files

### PR DESCRIPTION
**Pull request by spacecowboy**

### CODE REVIEW

**Comment by spacecowboy** [comment]
[Fine-grained control of filesystem locations (ie, tx logs)](https://trello.com/c/kXCTJvDN/325-fine-grained-control-of-filesystem-locations-ie-tx-logs)

File: enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/CausalClusteringSettings.java [hunk 1]
```diff
@@ -188,6 +192,11 @@
     public static final Setting<Boolean> raft_messages_log_enable =
             setting( "causal_clustering.raft_messages_log_enable", BOOLEAN, FALSE);
 
+    @Description( "Path to RAFT messages log." )
+    public static final Setting<File> raft_messages_log_path =
+            derivedSetting( "dbms.logs.query.path", logs_directory, ( logs ) -> new File( logs, "raft-messages.log" ),
+                    PATH );
+
```
Comment (davidegrohmann) [inline enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/CausalClusteringSettings.java]: Moreover the configuration name says "path", but isn't it the actual log file?

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java [hunk 2]
```diff
@@ -141,6 +144,11 @@ private ThirdPartyJaxRsPackage createThirdPartyJaxRsPackage( String packageAndMo
     @Description( "Enable HTTP request logging." )
     public static final Setting<Boolean> http_logging_enabled = setting( "dbms.logs.http.enabled", BOOLEAN, FALSE );
 
+    @Description( "Path to HTTP request log." )
+    public static final Setting<File> http_log_path =
+            derivedSetting( "dbms.logs.http.path", logs_directory, ( logs ) -> new File( logs, "http.log" ),
+                    PATH );
+
```
Comment (davidegrohmann) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:151]: Similarly here, we say "path" but it is the actual file no?

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 3]
```diff
@@ -502,6 +502,12 @@
             ( logs ) -> new File( logs, "query.log" ),
             PATH );
 
+    @Description( "Path to the debug log file." )
+    public static final Setting<File> store_internal_log_path = derivedSetting( "dbms.logs.debug.path",
+            logs_directory,
+            ( logs ) -> new File( logs, "debug.log" ),
+            PATH );
+
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:510]: Here as well, path vs file.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java [hunk 4]
```diff
@@ -41,7 +41,7 @@
 
 public class StoreLogService extends AbstractLogService implements Lifecycle
 {
-    public static final String INTERNAL_LOG_NAME = "debug.log";
+    //public static final String INTERNAL_LOG_NAME = "debug.log";
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/logging/StoreLogService.java]: left over?

File: community/kernel/src/test/java/org/neo4j/unsafe/batchinsert/internal/BatchInsertTest.java [hunk 5]
```diff
@@ -639,7 +638,7 @@ public void messagesLogGetsClosed() throws Exception
         File storeDir = this.storeDir.graphDbDir();
         BatchInserter inserter = BatchInserters.inserter( storeDir, fileSystemRule.get(), stringMap() );
         inserter.shutdown();
-        assertTrue( new File( storeDir, StoreLogService.INTERNAL_LOG_NAME ).delete() );
+        assertTrue( new File( storeDir, "debug.log" ).delete() );
```
Comment (davidegrohmann) [inline community/kernel/src/test/java/org/neo4j/unsafe/batchinsert/internal/BatchInsertTest.java]: GraphDatabaseInternalLogIT.INTERNAL_LOG_FILE instead?

File: enterprise/ha/src/test/java/org/neo4j/ha/TestBranchedData.java [hunk 6]
```diff
@@ -338,7 +337,7 @@ private long moveAwayToLookLikeOldBranchedDirectory() throws IOException
         for ( File file : Objects.requireNonNull( dir.listFiles() ) )
         {
             String fileName = file.getName();
-            if ( !fileName.equals( StoreLogService.INTERNAL_LOG_NAME ) && !file.getName().startsWith( "branched-" ) )
+            if ( !fileName.equals( "debug.log" ) && !file.getName().startsWith( "branched-" ) )
```
Comment (davidegrohmann) [inline enterprise/ha/src/test/java/org/neo4j/ha/TestBranchedData.java]: GraphDatabaseInternalLogIT.INTERNAL_LOG_FILE ?

**Review by davidegrohmann — APPROVED** [review]
Small comments

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java (hunk 2 above)
Comment (spacecowboy) [reply community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:151]: Yes, but a path can refer to both a file and a directory, right? Plus, we stay consistent with `dbms.logs.query.path` which already exists.

Comment (davidegrohmann) [reply community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:151]: It is not super important, I simply find confusing using the same term for 2 different things.

**Pull request merged**